### ISSUE TITLE
simple_multiset ignores the last element?

### ISSUE THREAD

**Issue by batterseapower** [dkondor/orbtree#1]
The example below demonstrates the problem:

```
struct count {
    typedef float argument_type;
    typedef unsigned int result_type;
    unsigned int operator()(const float& p) const {
        return 10;
    }
};

int main() {
    orbtree::simple_multiset<int, count> map;
    printf("%d %d %d\n", map.get_sum(5 + -1), map.get_sum(5), map.get_sum(5 + 1)); // 0 0 0

    // Adding this line fixes the tests below
    //map.insert(1000);

    map.insert(5);
    printf("%d %d %d\n", map.get_sum(5 + -1), map.get_sum(5), map.get_sum(5 + 1)); // 0 0 0: should be 0 0 10
    
    map.insert(5);
    printf("%d %d %d\n", map.get_sum(5 + -1), map.get_sum(5), map.get_sum(5 + 1)); /// 0 0 0: should be 0 0 20

    map.insert(5 + 2);
    printf("%d %d %d\n", map.get_sum(5 + -1), map.get_sum(5), map.get_sum(5 + 1)); // 0 0 20: should be 0 0 20
    
    printf("erase 2: %d\n", map.erase(5 + 2)); // 1
    printf("%d %d %d\n", map.get_sum(5 + -1), map.get_sum(5), map.get_sum(5 + 1)); // 0 0 0: should be 0 0 20

    map.insert(5 + -1);
    printf("%d %d %d\n", map.get_sum(5 + -1), map.get_sum(5), map.get_sum(5 + 1)); // 0 10 0: should be 0 10 30

    printf("erase -1: %d\n", map.erase(5 + -1)); // 1
    printf("%d %d %d\n", map.get_sum(5 + -1), map.get_sum(5), map.get_sum(5 + 1)); // 0 0 0: should be 0 0 20

    printf("erase 0: %d\n", map.erase(5)); // 2
    printf("%d %d %d\n", map.get_sum(5 + -1), map.get_sum(5), map.get_sum(5 + 1)); // 0 0 0
}
```

**Comment by dkondor** [comment]
Hi, thank you for reporting this and providing the example!
I've created a fix in the fix_sum branch in PR #2. Let me know if you have any further problems, otherwise, I'll merge it soon.

**Comment by batterseapower** [comment]
Thanks for the quick fix!
